drivers/ethernet/Kconfig.stm32_hal

```diff
@@ -1,14 +1,17 @@
 # STM32 HAL Ethernet driver configuration options
 
 # Copyright (c) 2017 Erwin Rol <erwin@erwinrol.com>
+# Copyright (c) 2020 Alexander Kozhinov <AlexanderKozhinov@yandex.com>
 # SPDX-License-Identifier: Apache-2.0
 
 menuconfig ETH_STM32_HAL
 	bool "STM32 HAL Ethernet driver"
 	select USE_STM32_HAL_ETH
+	select USE_STM32_HAL_RCC_EX if SOC_SERIES_STM32H7X
+	select NOCACHE_MEMORY if SOC_SERIES_STM32H7X
```

Comment:
NOCACHE_MEMORY depends (indirectly) on CPU_CORTEX_M7 arch, which restricts this driver implementation to M7 core on Dual Core H7 SoCs.
I'm fine with this, but this should be documented and verified in the code.

Reply:
Thanks for the info. You're right, `NOCACHE_MEMORY` makes sense only for M7 core, since it has DTCM coupled to it and M4 not. Any suggestions how `NOCACHE_MEMORY` can be coupled on M7 (e.g. select `NOCACHE_MEMORY if SOC_SERIES_STM32H7X && CPU_M7`)?

Reply:
`select NOCACHE_MEMORY if SOC_SERIES_STM32H7X && CPU_CORTEX_M7` should work.

Then, do you know if the driver will work on M4 core?
If not, it would be better to prohibit its use on M4 + H7 for now.

Reply:
I am pretty sure the driver will work on M4, but it is currently not tested. 
I would change then the line 11 as follows: `select NOCACHE_MEMORY if SOC_SERIES_STM32H7X && CPU_M7`
I will test now if `CPU_M7` Kconfig define or something similar is allowed. Will let you know within next new push.

Reply:
The line 11 changed to: `select NOCACHE_MEMORY if SOC_SERIES_STM32H7X && CPU_CORTEX_M7`